### PR TITLE
Add PGD 4.3.2 release notes

### PR DESCRIPTION
**Pull request by sunnyku**
## What Changed?

Fixes: BDR-3975

### CODE REVIEW

File: product_docs/docs/pgd/4/rel_notes/pgd_4.3.2_rel_notes.mdx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+EDB Postgres Distributed version 4.3.2 is a minor release of EDB Postgres Distributed 4, which includes bug fixes for issues identified in previous versions.
+
+!!! Note
+    This version is required for EDB Postgres Advanced Server versions 12.15, 13.11, 14.8 and later.
+
+| Component | Version |  Type           |                                                   Description                                                           |
+| --------- | ------- | --------------- | ------------------------------------------------------------------------------------------------------------------------|
+| BDR       | 4.3.2   | Feature         | Fix a bug in handling of memory context in presence of triggers |
+| BDR       | 4.3.2   | Feature         | Add some simplistic memory tracking util |
```
Comment (pavanvd) [inline product_docs/docs/pgd/4/rel_notes/pgd_4.3.2_rel_notes.mdx:14]: This is an internal utility, not even visible in customer builds. Please remove this

File: product_docs/docs/pgd/4/rel_notes/pgd_4.3.2_rel_notes.mdx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+| BDR       | 4.3.2   | Feature         | Add a test case for bulkload with triggers on downstream |
```
Comment (pavanvd) [inline product_docs/docs/pgd/4/rel_notes/pgd_4.3.2_rel_notes.mdx:15]: This is not a feature and maybe not even worth mentioning in the release notes. I don't think users get access to our regression tests

File: product_docs/docs/pgd/4/rel_notes/pgd_4.3.2_rel_notes.mdx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+| BDR       | 4.3.2   | Feature         | Prevent overflow of EagerTxnConfig::node_idss array |
+| BDR       | 4.3.2   | Feature         | Add upgrade path from 3.7.22 |
+| BDR       | 4.3.2   | Feature         | Fixup bdr.sequence_alloc upgrade script |
+| BDR       | 4.3.2   | Feature         | Check for malformed connection string |
```
Comment (pavanvd) [inline product_docs/docs/pgd/4/rel_notes/pgd_4.3.2_rel_notes.mdx:19]: I would recommend mentioning user visible change here. `bdr.create_node()` will fail on malformed connection string. Maybe also reference the customer ticket.

File: product_docs/docs/pgd/4/rel_notes/pgd_4.3.2_rel_notes.mdx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+!!! Note
+    This version is required for EDB Postgres Advanced Server versions 12.15, 13.11, 14.8 and later.
+
+| Component | Version |  Type           |                                                   Description                                                           |
+| --------- | ------- | --------------- | ------------------------------------------------------------------------------------------------------------------------|
+| BDR       | 4.3.2   | Feature         | Fix a bug in handling of memory context in presence of triggers |
+| BDR       | 4.3.2   | Feature         | Add some simplistic memory tracking util |
+| BDR       | 4.3.2   | Feature         | Add a test case for bulkload with triggers on downstream |
+| BDR       | 4.3.2   | Feature         | Prevent overflow of EagerTxnConfig::node_idss array |
```
Comment (pavanvd) [inline product_docs/docs/pgd/4/rel_notes/pgd_4.3.2_rel_notes.mdx:16]: Isn't too internal?

File: product_docs/docs/pgd/4/rel_notes/pgd_4.3.2_rel_notes.mdx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+| BDR       | 4.3.2   | Feature         | Add upgrade path from 3.7.22 |
+| BDR       | 4.3.2   | Feature         | Fixup bdr.sequence_alloc upgrade script |
+| BDR       | 4.3.2   | Feature         | Check for malformed connection string |
+| BDR       | 4.3.2   | Feature         | Remove txn_config entry from ReorderBuffer hash table |
+| BDR       | 4.3.2   | Feature         | Update bdr.sequence_alloc to match the C code |
+| BDR       | 4.3.2   | Feature         | Allow to delete the datadirs of failed TAP Tests |
```
Comment (pavanvd) [inline product_docs/docs/pgd/4/rel_notes/pgd_4.3.2_rel_notes.mdx:22]: Again, not worth adding to release notes

File: product_docs/docs/pgd/4/rel_notes/pgd_4.3.2_rel_notes.mdx [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+| BDR       | 4.3.2   | Feature         | Resetting MessageContext in writer frees SubMessageContext |
+| BDR       | 4.3.2   | Feature         | fix init_install.sql test failure for witness node |
+| BDR       | 4.3.2   | Feature         | Ignore global_lock check from repset_func when SDW enabled |
+| BDR       | 4.3.2   | Feature         | PA committer signals next writer twice to enforce commit order |
```
Comment (pavanvd) [inline product_docs/docs/pgd/4/rel_notes/pgd_4.3.2_rel_notes.mdx:26]: "PA" is not something we've documented for users, right?

File: product_docs/docs/pgd/4/rel_notes/pgd_4.3.2_rel_notes.mdx [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+| BDR       | 4.3.2   | Feature         | Initialize cs_id of begindata in receiver |
+| BDR       | 4.3.2   | Feature         | Fix for BdrDb.pm |
```
Comment (pavanvd) [inline product_docs/docs/pgd/4/rel_notes/pgd_4.3.2_rel_notes.mdx:28]: What is the fix? :-)

File: product_docs/docs/pgd/4/rel_notes/pgd_4.3.2_rel_notes.mdx [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+| BDR       | 4.3.2   | Feature         | Add check for conflicting node names |
+| BDR       | 4.3.2   | Feature         | Eliminate errors from new gcc12 warnings |
```
Comment (pavanvd) [inline product_docs/docs/pgd/4/rel_notes/pgd_4.3.2_rel_notes.mdx:30]: Not definitely a feature and probably not worth adding to release notes either

**Comment by sunnyku** [comment]
Thanks for the review @pavanvd but it was not ready for review. it is now.

**Comment by sunnyku** [comment]
Hey @djw-m,  re-based to pick new date format. Thanks :+1:

**Comment by LenzGr** [comment]
This was raised by @kelpoole - please ensure to also update the [supported upgrade paths document](https://www.enterprisedb.com/docs/pgd/4/upgrades/upgrade_paths/) as part of this PR. Thanks!
Please also take note of this step in the release checklist.

File: product_docs/docs/pgd/4/rel_notes/pgd_4.3.2_rel_notes.mdx [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+| --------- | ------- | --------------- | ------------------------------------------------------------------------------------------------------------------------|
+| BDR       | 4.3.2   | Bug fix         | Fixed a bug in handling of memory context in presence of triggers. |
+| BDR       | 4.3.2   | Bug fix         | Prevent overflow of EagerTxnConfig::node_idss array |
+| BDR       | 4.3.2   | Bug fix         | Fixed `bdr.sequence_alloc` upgrade script. |
+| BDR       | 4.3.2   | Bug fix         | Fixed check for malformed connection string tp prevent failure in `bdr.create_node()`. (RT95453) |
+| BDR       | 4.3.2   | Bug fix         | Remove txn_config entry from ReorderBuffer hash table. |
+| BDR       | 4.3.2   | Bug fix         | Fixed a memory leak in `bdr.sequence_alloc`. |
+| BDR       | 4.3.2   | Bug fix         | Ignore global_lock check from repset_func when SDW enabled |
+| BDR       | 4.3.2   | Bug fix         | Added check for conflicting node names. |
+| BDR       | 4.3.2   | Bug fix         | Added wrapper for checking RTE permissions to disable pgaudit |
```
Comment (PJMODOS) [inline product_docs/docs/pgd/4/rel_notes/pgd_4.3.2_rel_notes.mdx:21]: "Add workaround for pgaudit crash bug 212 which occurs with some 3rd party extensions, including BDR"

File: product_docs/docs/pgd/4/rel_notes/pgd_4.3.2_rel_notes.mdx [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+
+EDB Postgres Distributed version 4.3.2 is a minor release of EDB Postgres Distributed 4, which includes bug fixes for issues identified in previous versions.
+
+!!! Note
+    This version is required for EDB Postgres Advanced Server versions 12.15, 13.11, 14.8 and later.
+
+| Component | Version |  Type           |                                                   Description                                                           |
+| --------- | ------- | --------------- | ------------------------------------------------------------------------------------------------------------------------|
+| BDR       | 4.3.2   | Bug fix         | Fixed a bug in handling of memory context in presence of triggers. |
+| BDR       | 4.3.2   | Bug fix         | Prevent overflow of EagerTxnConfig::node_idss array |
```
Comment (PJMODOS) [inline product_docs/docs/pgd/4/rel_notes/pgd_4.3.2_rel_notes.mdx:14]: "Prevent overflow in group commit transaction tracking on large clusters"

File: product_docs/docs/pgd/4/rel_notes/pgd_4.3.2_rel_notes.mdx [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+---
+
+EDB Postgres Distributed version 4.3.2 is a minor release of EDB Postgres Distributed 4, which includes bug fixes for issues identified in previous versions.
+
+!!! Note
+    This version is required for EDB Postgres Advanced Server versions 12.15, 13.11, 14.8 and later.
+
+| Component | Version |  Type           |                                                   Description                                                           |
+| --------- | ------- | --------------- | ------------------------------------------------------------------------------------------------------------------------|
+| BDR       | 4.3.2   | Bug fix         | Fixed a bug in handling of memory context in presence of triggers. |
```
Comment (PJMODOS) [inline product_docs/docs/pgd/4/rel_notes/pgd_4.3.2_rel_notes.mdx:13]: "Fix crash in conflict triggers during bulk inserts"

File: product_docs/docs/pgd/4/rel_notes/pgd_4.3.2_rel_notes.mdx [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+| BDR       | 4.3.2   | Bug fix         | Prevent overflow of EagerTxnConfig::node_idss array |
+| BDR       | 4.3.2   | Bug fix         | Fixed `bdr.sequence_alloc` upgrade script. |
+| BDR       | 4.3.2   | Bug fix         | Fixed check for malformed connection string tp prevent failure in `bdr.create_node()`. (RT95453) |
```
Comment (PJMODOS) [inline product_docs/docs/pgd/4/rel_notes/pgd_4.3.2_rel_notes.mdx:16]: Validate node connection strings in `bdr.create_node()`

File: product_docs/docs/pgd/4/rel_notes/pgd_4.3.2_rel_notes.mdx [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+| BDR       | 4.3.2   | Bug fix         | Remove txn_config entry from ReorderBuffer hash table. |
+| BDR       | 4.3.2   | Bug fix         | Fixed a memory leak in `bdr.sequence_alloc`. |
```
Comment (PJMODOS) [inline product_docs/docs/pgd/4/rel_notes/pgd_4.3.2_rel_notes.mdx:18]: There is no such thing in this release

File: product_docs/docs/pgd/4/rel_notes/pgd_4.3.2_rel_notes.mdx [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+| BDR       | 4.3.2   | Bug fix         | Ignore global_lock check from repset_func when SDW enabled |
```
Comment (PJMODOS) [inline product_docs/docs/pgd/4/rel_notes/pgd_4.3.2_rel_notes.mdx:19]: This is test change, not user visible

File: product_docs/docs/pgd/4/rel_notes/pgd_4.3.2_rel_notes.mdx [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+| BDR       | 4.3.2   | Bug fix         | Added check for conflicting node names. |
+| BDR       | 4.3.2   | Bug fix         | Added wrapper for checking RTE permissions to disable pgaudit |
+| BDR       | 4.3.2   | Bug fix         | Fixed the consensus snapshot compatibility with PGD 3.7. (RT93022) |
+| BDR       | 4.3.2   | Bug fix         | Fixed an issue by allowing a logical join of node if there are foreign key constraints violations. (RT91745) |
```
Comment (PJMODOS) [inline product_docs/docs/pgd/4/rel_notes/pgd_4.3.2_rel_notes.mdx:23]: "Don't check foreign keys during node join, this makes the join behave the same as replication"

File: product_docs/docs/pgd/4/rel_notes/pgd_4.3.2_rel_notes.mdx [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+| BDR       | 4.3.2   | Bug fix         | Foreign key constraints will be skipped during sync. |
```
Comment (PJMODOS) [inline product_docs/docs/pgd/4/rel_notes/pgd_4.3.2_rel_notes.mdx:24]: This si duplicate of above

File: product_docs/docs/pgd/4/rel_notes/pgd_4.3.2_rel_notes.mdx [hunk 17]
```diff
@@ -0,0 +1,26 @@
...
+| BDR       | 4.3.2   | Bug fix         | Writers wakeup receiver waiting for free commit queue entry |
```
Comment (PJMODOS) [inline product_docs/docs/pgd/4/rel_notes/pgd_4.3.2_rel_notes.mdx:25]: Improve signalling between writers and receiver when the writer queue is full

**Comment by PJMODOS** [comment]
I rewrote some of the notes, please make sure the branch is squished before merging

**Comment by sunnyku** [comment]
> I rewrote some of the notes, please make sure the branch is squished before merging

I was about to repush with suggested modifications. Thanks :+1:

File: product_docs/docs/pgd/4/rel_notes/index.mdx [hunk 18]
```diff
@@ -25,6 +26,7 @@ The EDB Postgres Distributed documentation describes the latest version of EDB P
 
 | Release Date | EDB Postgres Distributed     | BDR   | HARP  |  CLI  | TPAexec                                                                          |
 | ------------ | ---------------------------- | ----- | ----- | ----- | -------------------------------------------------------------------------------- |
+| 30 Aug 2023 | [4.3.2 ](pgd_4.3.2_rel_notes)| 4.3.2 | 2.3.1 | 1.1.1 | [23.19](/tpa/latest/rel_notes/tpa_23.19_rel_notes)   |
```
Comment (LenzGr) [inline product_docs/docs/pgd/4/rel_notes/index.mdx:29]:
```suggestion
| 30 Aug 2023 | [4.3.2 ](pgd_4.3.2_rel_notes)| 4.3.2 | 2.3.1 | 1.1.1 | [23.20](/tpa/latest/rel_notes/tpa_23.20_rel_notes/)   |
```

Comment (LenzGr) [reply product_docs/docs/pgd/4/rel_notes/index.mdx:29]: This version change was pointed out by @kelpoole

**Review by LenzGr — DISMISSED** [review]
Looks good, thank you!

**Pull request merged**